modules/core/src/passes/screen-pass-uniforms.ts

```diff
@@ -0,0 +1,27 @@
+import type {TextureView} from '@luma.gl/core';
+import {ShaderModule} from '@luma.gl/shadertools';
+import {UniformTypes} from '../shaderlib/misc/uniform-types';
+
+const uniformBlock = `\
+uniform screenUniforms {
+  vec2 texSize;
+} screen;
+`;
+
+type ScreenBindingProps = {
+  texSrc: TextureView;
+};
+
+type ScreenUniformProps = {
+  texSize: [number, number];
+};
+
+export type ScreenProps = ScreenBindingProps & ScreenUniformProps;
+
+export const screenUniforms = {
+  name: 'screen',
+  fs: uniformBlock,
+  uniformTypes: {
+    texSize: 'vec2<f32>'
+  } as const satisfies UniformTypes<Required<ScreenUniformProps>>
```

Comment:
I wonder if we should review the `ShaderModule` type, the `UniformsT` & `BindingsT` [generics](https://github.com/visgl/luma.gl/blob/4d03a8d57245bd4c104a8e357a2dad13394a9284/modules/shadertools/src/lib/shader-module/shader-module.ts#L27-L28) don't seem to be that useful IMHO. I think it would be better to have the `ShaderModule` be generic on `PropsT` and `UniformTypesT`, as `UniformsT` and `BindingsT` can be obtained by filtering `PropsT`.

I've had a play and we could remove the nested `const satisfies UniformTypes<..>` by using the following filtering type helper:

```ts
type UniformValue = number | boolean | number[];
type FilterUniformKeys<T> = {[K in keyof T]: T[K] extends UniformValue ? K : never}[keyof T];
type UniformsOnly<T> = {[K in FilterUniformKeys<T>]: T[K]};
```

We then don't need the `ScreenBindingProps` & `ScreenUniformProps` types and can just do:

```ts
export const screenUniforms = {
  name: 'screen',
  fs: uniformBlock,
  uniformTypes: {
    texSize: 'vec2<f32>'
  } as const satisfies UniformTypes<UniformsOnly<ScreenProps>>
} as const satisfies ShaderModule<ScreenProps>;

```

Which could all be encapsulated in luma's `ShaderModule` definition